python/ray/data/tests/test_dataset_formats.py

```diff
@@ -327,13 +327,11 @@ def test_parquet_read_partitioned(ray_start_regular_shared, fs, data_path):
     assert len(input_files) == 2, input_files
     assert (
         str(ds) == "Dataset(num_blocks=2, num_rows=6, "
-        "schema={two: string, "
-        "one: dictionary<values=int32, indices=int32, ordered=0>})"
+        "schema={two: string, one: int64})"
```

Comment:
Hm why did it change from dict to int?

Reply:
The Arrow schema produced when [reading Parquet metadata files](https://arrow.apache.org/docs/python/parquet.html#reading-from-partitioned-datasets) will represent partition columns as Arrow dictionary types (Arrow analogue for Pandas categorical):

>Note: the partition columns in the original table will have their types converted to Arrow dictionary types (pandas categorical) on load.

I believe that this column type is the post-read type now since we're now always fetching the metadata from the first read task, see [this comment](https://github.com/ray-project/ray/pull/23821/files#r847458253).

Reply:
Btw, I'm realizing that we might want to go back to relying on pre-read metadata for datasources like Parquet, so I'll look at adding back the old logic of "only fetch metadata from read task if we don't have pre-read schema".